src/API/Management/Roles.php

```diff
@@ -0,0 +1,307 @@
+<?php
+
+namespace Auth0\SDK\API\Management;
+
+use Auth0\SDK\Exception\CoreException;
+
+/**
+ * Class Roles.
+ * Handles requests to the Roles endpoint of the v2 Management API.
+ *
+ * @package Auth0\SDK\API\Management
+ */
+class Roles extends GenericResource
+{
+    /**
+     * Get all Roles
+     * Required scope: "read:roles"
+     *
+     * @param array $params Additional parameters to send with the request.
+     *
+     * @return mixed
+     *
+     * @throws \Exception Thrown by the HTTP client when there is a problem with the API call.
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Roles/get_roles
+     */
+    public function getAll(array $params = [])
+    {
+        return $this->apiClient->method('get')
+            ->withDictParams($this->normalizePagination( $params ))
+            ->addPath('roles')
+            ->call();
+    }
+
+    /**
+     * Create a new Role.
+     * Required scope: "create:roles"
+     *
+     * @param string $name Role name.
+     * @param array  $data Additional fields to add, like description.
+     *
+     * @return mixed
+     *
+     * @throws CoreException Thrown if the identifier parameter or data field is empty or is not a string.
+     * @throws \Exception Thrown by the HTTP client when there is a problem with the API call.
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Roles/post_roles
+     */
+    public function create($name, array $data = [])
+    {
+        $data['name'] = (string) $name;
+
+        if (empty($data['name'])) {
+            throw new CoreException('Missing required name parameter.');
+        }
+
+        return $this->apiClient->method('post')
+            ->addPath('roles')
+            ->withBody(json_encode($data))
+            ->call();
+    }
+
+    /**
+     * Get a single Role by ID.
+     * Required scope: "read:roles"
+     *
+     * @param string $role_id Role ID to get.
+     *
+     * @return mixed
+     *
+     * @throws CoreException Thrown if the id parameter is empty or is not a string.
+     * @throws \Exception Thrown by the HTTP client when there is a problem with the API call.
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Roles/get_roles_by_id
+     */
+    public function get($role_id)
+    {
+        if (empty($role_id) || ! is_string($role_id)) {
```

Comment:
Not very DRY how this is repeated in so many functions.  Maybe refactor it out to a simple function in case you want to change the exception method or implement il8n later?